src/gsy_e/models/strategy/predefined_pv.py

```diff
@@ -92,9 +92,9 @@ def __init__(
     def read_config_event(self):
         # this is to trigger to read from self.simulation_config.cloud_coverage:
         self.cloud_coverage = None
-        self.set_produced_energy_forecast_kWh_future_markets(reconfigure=True)
+        self.set_produced_energy_forecast_kWh_in_state(reconfigure=True)
 
-    def set_produced_energy_forecast_kWh_future_markets(self, reconfigure=True):
+    def set_produced_energy_forecast_kWh_spot_market(self, reconfigure=True):
```

Comment:
You need to reimplement `set_produced_energy_forecast_kWh_future_market` for the predefined PV, too, otherwise the gaussian profile will be saved for the future markets.
Another way would be to loop here again over the list of `[spot_market_slot, future_market_slot_1, future_market_slot_2,]` and call the `self.state.set_available_energy(available_energy_kWh, slot_time, reconfigure)
` for each

Reply:
Thanks a lot for the proposal! I have adapted [here](https://github.com/gridsingularity/gsy-e/pull/1389/commits/e347f7e4505b98071c264021d85d12479eb03a6f)